spec/syntax.md

```diff
@@ -438,55 +445,65 @@ text-char = %x0-5B         ; omit \
           / %xE000-10FFFF
 ```
 
+### Literals
+
+_Literal_ is used for matching variants and providing input to _expressions_.
+_Quoted_ literals may include content with any Unicode code point,
+except for surrogate code points U+D800 through U+DFFF.
+The characters `\` and `|` MUST be escaped as `\\` and `\|`.
+_Unquoted_ literals have a much more restricted range that
+is intentionally close to the XML's [Nmtoken](https://www.w3.org/TR/xml/#NT-Nmtoken),
+with the restriction that it MUST NOT start with `-` or `:`,
+as those would conflict with _function_ start characters.
+
+All code points are preserved.
+
 ```abnf
-literal = "|" *(literal-char / literal-escape) "|"
-literal-char = %x0-5B         ; omit \
-             / %x5D-7B        ; omit |
-             / %x7D-D7FF      ; omit surrogates
-             / %xE000-10FFFF
+literal = quoted / unquoted
+
+quoted         = "|" *(quoted-char / quoted-escape) "|"
+quoted-char    = %x0-5B         ; omit \
+               / %x5D-7B        ; omit |
+               / %x7D-D7FF      ; omit surrogates
+               / %xE000-10FFFF
+
+unquoted       = unquoted-start *name-char
+unquoted-start = name-start / DIGIT / "."
+               / %xB7 / %x300-36F / %x203F-2040
 ```
 
 ### Names
 
 The _name_ token is used for variable names (prefixed with `$`),
 function names (prefixed with `:`, `+` or `-`),
 as well as option names.
-A name MUST NOT start with an ASCII digit and certain basic combining characters.
+It is based on XML's [Name](https://www.w3.org/TR/xml/#NT-Name),
+with the restriction that it MUST NOT start with an ASCII digit and certain basic combining characters.
```

Comment:
The phrasing here suggested that XML's `Name` allows digits as the first character.

Reply:
Good point. The only actual divergence is the colon.

```suggestion
with the restriction that it MUST NOT start with `:`,
as that would conflict with _function_ start characters.
```